src/puppetlabs/trapperkeeper/services/status/status_proxy_service.clj

```diff
@@ -1,22 +1,15 @@
 (ns puppetlabs.trapperkeeper.services.status.status-proxy-service
   (:require [clojure.tools.logging :as log]
-    [puppetlabs.trapperkeeper.core :refer [defservice]])
-  (:import (java.net URL)))
+    [puppetlabs.trapperkeeper.services.status.status-core :refer [get-proxy-route-info]]
+    [puppetlabs.trapperkeeper.core :refer [defservice]]))
 
 (defservice status-proxy-service
   [[:WebroutingService add-proxy-route]
    [:ConfigService get-in-config]]
   (init [this context]
     (log/info "Initializing status service proxy")
-    (let [target-url (URL. (get-in-config [:status-proxy :proxy-target-url]))
-          host (.getHost target-url)
-          port (.getPort target-url)
-          path (.getPath target-url)]
-      (add-proxy-route
-        this
-        {:host host
-         :port port
-         :path path}
-        {:ssl-config (get-in-config [:status-proxy :ssl-opts])
-         :scheme     :https}))
+    (let [status-proxy-config (get-in-config [:status-proxy])
+          {proxy-target :proxy-target
+           proxy-options :proxy-options} (get-proxy-route-info status-proxy-config)]
```

Comment:
Wow, that is some Clojure voodoo I did not know existed.

You can use the `:keys` destructuring magic here, it's a bit more concise and common:
```clj
    (let ...
         {:keys [proxy-target proxy-options]} (get-proxy-route-info status-proxy-config)]
      (add-proxy-route this proxy-target proxy-options))
```